trieste/models/gpflux/builders.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This file contains builders for GPflux models supported in Trieste. We found the default
+configurations used here to work well in most situation, but they should not be taken as
+universally good solutions.
+"""
+
+from __future__ import annotations
+
+from typing import Optional
+
+import gpflow
+import numpy as np
+import tensorflow as tf
+from gpflux.architectures import Config, build_constant_input_dim_deep_gp
+from gpflux.models import DeepGP
+
+from ...data import Dataset
+from ...space import Box, SearchSpace
+from ...types import TensorType
+
+
+def build_vanilla_deep_gp(
+    data: Dataset,
+    num_layers: int = 2,
+    num_inducing_points: Optional[int] = None,
+    search_space: Optional[SearchSpace] = None,
+    inner_layer_sqrt_factor: float = 1e-5,
+    likelihood_variance: float = 1e-2,
+    trainable_likelihood: bool = True,
```

Comment:
@sebastianober we recently created similar build functions for gpflow models and we tried to give sensible default values for everything - so I'm making this function more consistent with those, e.g. using data to set the kernel variance and mean function and so on, please see builders.py file in gpflow, but all the values here would welcome your input